src/classes/CountryStateDistrictMapDataManipulation.cls

```diff
@@ -0,0 +1,201 @@
+    *                        V1.1  - Modified -   14/07/2022    
+ */
+
+public with sharing class CountryStateDistrictMapDataManipulation {
+    // Global Map
+    Map<String, Map<String, List<String>>> mapOfCountryWithStateAndDistrict = new Map<String, Map<String, List<String>>>();
+    // Class Parameterized Constructor
+    public CountryStateDistrictMapDataManipulation(String firstLetter) {
+        
+        List<District__c> districtList = [Select Name, State_Name__r.Name, State_Name__r.Country_Name__r.Name From District__c];
+        List<String> combinationOfCountryStateAndDistrict = new list<String>();
+        If(districtList.size() > 0) {
+            for(District__c district : districtList) {
+                If(!mapOfCountryWithStateAndDistrict.containsKey(district.State_Name__r.Country_Name__r.Name)) {
+                    mapOfCountryWithStateAndDistrict.put(district.State_Name__r.Country_Name__r.Name, new Map<String, List<String>>());
+                }
+                If(!mapOfCountryWithStateAndDistrict.get(district.State_Name__r.Country_Name__r.Name).containsKey(district.State_Name__r.Name)) {
+                    mapOfCountryWithStateAndDistrict.get(district.State_Name__r.Country_Name__r.Name).put(district.State_Name__r.Name, new List<String>());
+                }
+                If(!mapOfCountryWithStateAndDistrict.get(district.State_Name__r.Country_Name__r.Name).get(district.State_Name__r.Name).contains(district.Name)) {
+                    mapOfCountryWithStateAndDistrict.get(district.State_Name__r.Country_Name__r.Name).get(district.State_Name__r.Name).add(district.Name);
+                    If(district.State_Name__r.Country_Name__r.Name.startsWithIgnoreCase(firstLetter)) {
+                        If(district.State_Name__r.Name.startsWithIgnoreCase(firstLetter)) {
+                            If(district.Name.startsWithIgnoreCase(firstLetter)) {
+                                combinationOfCountryStateAndDistrict.add(district.State_Name__r.Country_Name__r.Name + ' -> ' + district.State_Name__r.Name + ' -> ' + district.Name);
+                            }
+                        }
+                    }
+                }
+            }
+        }
+        System.debug('Map of Country, State and District ==> ' + mapOfCountryWithStateAndDistrict);
+        System.debug('Combination of Country, State and District ==> ' + combinationOfCountryStateAndDistrict); 
+        findTotalStateAndDistrict();        
+    }
+
+    /* 
+        *    @Description    :      Inside this method we update the total districts, total states of country & states using Aggregate query.
+        *
+        *    @args           :       
+        *
+        *    @return         :       void (Nohting)
+        *
+        *    @revision Log   :       V1.1  - Created  -  13/07/2022  - Shubham Kumar  - Used for Practise
+        *                                    
+    */
+
+    // Method
+    public static void findTotalStateAndDistrict() {
```

Comment:
Global map filling concept  + AggregateResult Concept will be in Constructor and finding the country, state, and District logic will be in the Parameterized Constructor.